File: Solutions/8938/c++/program.cpp
// Copy from Quera

#include <iostream>

using namespace std;

int main()
{
    int n, m;
    cin >> n >> m;

    int a[n][n];
    for (int i = 0; i < n; i++)
        for (int j = 0; j < n; j++)
            cin >> a[i][j];

    int ans = 0;
    for (int i = 0; i < m; i++)
    {
        int x, y;
        cin >> x >> y;
        ans += a[x - 1][y - 1];
    }

    cout << ans << '\n';
    return 0;
}

File: Solutions/8938/java/program.java
// Copy from Quera

import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt(), m = scanner.nextInt();
        
        int [][] a = new int[n][n];
        for (int i = 0; i < n; i++)
            for (int j = 0; j < n; j++)
                a[i][j] = scanner.nextInt();

        int ans = 0;
        for (int i = 0; i < m; i++) {
            int x = scanner.nextInt(), y = scanner.nextInt();
            ans += a[x - 1][y - 1];
        }

        System.out.println(ans);
    }
}